### ISSUE TITLE
Flutter master builds flutter-1.19-candidate.2 and later not working on Git Bash for Windows. Shows: Waiting for another flutter command to release the startup lock...

### ISSUE THREAD

**Issue by szepetry** [flutter/flutter#58993]
## Steps to Reproduce

1. Run Git bash terminal of Windows.
2. Do a flutter upgrade. (versions with issue: (flutter-1.19-candidate.2, flutter-1.19-candidate.3 and flutter-1.19-candidate.4)
3. Run flutter doctor. (or any flutter command including commands like flutter channel stable, etc.)
4. You'd get 'Waiting for another flutter command to release startup lock...'

## The issue

I am aware of the statement 'Waiting for another flutter command to release startup lock...' which dart leaves when a process is going on. I have tried all the fixes for that including deleting the lockfile in 'flutter/bin/cache' and killing all dart processes and the IDEs using it. This seems to be a different issue. This problem only comes on Git Bash for Windows but everything works on other terminals. I have currently downgraded to the flutter-1.19-candidate.1 master build by doing a git checkout to flutter-1.19-candidate.1. 

## TL:DR version
1. Current master builds flutter-1.19-candidate.2 , 3 and 4 aren't working on Git bash for Windows. 
2. On executing any flutter commands on Git bash I'd get 'Waiting for another flutter command to release startup lock...'
3. Tried the fixes for the 'Waiting for another flutter command to release startup lock...' issue as mentioned in the above section. 
4. The builds work on other terminals like cmd and ps.

## flutter doctor -v for flutter-1.19-candidate.1
![image](https://user-images.githubusercontent.com/24851270/84055273-383d7700-a9d2-11ea-90f0-8678847740ee.png)

**Comment by TahaTesser** [comment]
Hi @szepetry 
Can you please upgrade to latest `master` and try again?

`flutter channel master`
`flutter upgrade`
`flutter doctor -v`

Looks like you've a 4 weeks old flutter channel
Thank you

**Comment by szepetry** [comment]
> Hi @szepetry
> Can you please upgrade to latest `master` and try again?
> 
> `flutter channel master`
> `flutter upgrade`
> `flutter doctor -v`
> 
> Looks like you've a 4 weeks old flutter channel
> Thank you

@TahaTesser , Thank you for your response.

This is the `flutter doctor -v` with the current updated master build. Executed in cmd since my Git bash isn't working.
![image](https://user-images.githubusercontent.com/24851270/84196376-baef3080-aabd-11ea-9ca1-9a1a19f7bdec.png)

I had to use the old master build because the issue didn't occur on that. I can't use stable build because there is a maps issue on that right now which hasn't been fixed in stable. 

This is what comes up in my Git bash when I try to run flutter commands in it.
![image](https://user-images.githubusercontent.com/24851270/84196613-191c1380-aabe-11ea-80f0-4caf70cd8fa4.png)

This didn't happen in the master build flutter-1.19-candidate.1.

**Comment by jmagman** [comment]
\cc @gspencergoog

**Comment by jmagman** [comment]
Didn't think it ever worked in git bash.

**Comment by szepetry** [comment]
I have been using flutter in Git bash for over an year and it has always worked flawlessly for me.

**Comment by szepetry** [comment]
Alright, thanks for clearing it out. I am curious about how it could stop working. If you have an idea about how that happened, let me know. Thankss 😁

**Comment by christopherfujino** [comment]
> Alright, thanks for clearing it out. I am curious about how it could stop working. If you have an idea about how that happened, let me know. Thankss

There was a recent refactor to our folder-locking logic that didn't take into account git bash: https://github.com/flutter/flutter/pull/57590. I believe the author is looking into this further.

**Comment by gspencergoog** [comment]
So, the quick workaround for this is:

```
alias flutter=flutter.bat
```

That will still do proper locking on Windows, and avoid this problem.

The root of the problem is that git-bash doesn't have `flock` (or `shlock`), so it falls back to using an atomically created directory, but it's Windows, so the native file locking gets in the way sometimes of removing the lock directory.

I'm looking into adding a clause in the flutter bash script to instead exec the batch script when we're on Windows, so you won't need to alias it.

**Comment by gspencergoog** [comment]
As far as I can tell, #59789 fixes this.  Closing.
